test/runtests.jl

```diff
@@ -220,23 +220,23 @@ end
     )
 
     @test_throws(
-        ErrorException("unmanaged type in ref resolution $(Int64): 1."),
+        MethodError,
```

Comment:
I don't like this error. It doesn't offer any insight into what went wrong. I'll take a look at JSONPointer to see how it could be improved.

```Julia
julia> Schema("""{
                   "type": "object",
                   "properties": {"version": {"\$ref": "#/definitions/Foo"}},
                   "definitions": 1
               }""")
ERROR: MethodError: no method matching getindex(::Int64, ::SubString{String})
Closest candidates are:
  getindex(::Number) at number.jl:75
  getindex(::Number, ::Integer) at number.jl:76
  getindex(::Number, ::Integer...) at number.jl:81
  ...
Stacktrace:
 [1] getindex_by_pointer(::Dict{String,Any}, ::Tuple{SubString{String},SubString{String}}) at /Users/oscar/.julia/packages/JSONPointer/SXKBB/src/pointer.jl:185
 [2] getindex_by_pointer(::Dict{String,Any}, ::JSONPointer.Pointer{Any}) at /Users/oscar/.julia/packages/JSONPointer/SXKBB/src/pointer.jl:178
 [3] getindex(::Dict{String,Any}, ::JSONPointer.Pointer{Any}) at /Users/oscar/.julia/packages/JSONPointer/SXKBB/src/pointer.jl:110
 [4] find_ref(::HTTP.URIs.URI, ::Dict{String,Any}, ::String, ::String) at /Users/oscar/.julia/dev/JSONSchema/src/schema.jl:35
 [5] resolve_refs!(::Dict{String,Any}, ::HTTP.URIs.URI, ::Dict{String,Any}, ::String) at /Users/oscar/.julia/dev/JSONSchema/src/schema.jl:98
 [6] resolve_refs!(::Dict{String,Any}, ::HTTP.URIs.URI, ::Dict{String,Any}, ::String) at /Users/oscar/.julia/dev/JSONSchema/src/schema.jl:100 (repeats 2 times)
 [7] Schema(::Dict{String,Any}; parent_dir::String, parentFileDirectory::Nothing) at /Users/oscar/.julia/dev/JSONSchema/src/schema.jl:166
 [8] Schema(::Dict{String,Any}) at /Users/oscar/.julia/dev/JSONSchema/src/schema.jl:160
 [9] Schema(::String; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /Users/oscar/.julia/dev/JSONSchema/src/schema.jl:196
 [10] Schema(::String) at /Users/oscar/.julia/dev/JSONSchema/src/schema.jl:196
 [11] top-level scope at REPL[8]:1
```